app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -188,26 +187,38 @@ export default class ComposerEditor extends Component {
 
   @on("didInsertElement")
   _composerEditorInit() {
-    const input = this.element.querySelector(".d-editor-input");
     const preview = this.element.querySelector(".d-editor-preview-wrapper");
+    this._registerImageAltTextButtonClick(preview);
+
+    if (this.composer.allowUpload) {
+      this.uppyComposerUpload.setup(this.element);
+    }
+
+    this.appEvents.trigger(`${this.composerEventPrefix}:will-open`);
+  }
+
+  @bind
+  setupEditor(textManipulation) {
+    this.textManipulation = textManipulation;
+
+    const input = this.element.querySelector(".d-editor-input");
 
     input?.addEventListener(
       "scroll",
       this._throttledSyncEditorAndPreviewScroll
     );
```

Comment:
I know not related to the PR, but I think this logic should have its own modifier

Reply:
we can't use a proper `modifier` _here_ because the textarea is defined on `TextareaEditor`, that's why we're "prop-drilling" these lifecycle `onSetup` hooks... the idea is to add/remove the events whenever the `editorComponent` is switched with its own `TextManipulation` implementation

this `onSetup` call is just focusing the input (`putCursorAtEnd), hooking the scroll listener, and keeping `this.textManipulation` around to focus the input again whenever `this.composer.focusTarget` changes to "editor"

Reply:
No I think that's fine no hurries, as long as we keep doing progress like you do Im happy! Was just noting is as I saw it.